tools/autograd/templates/Functions.cpp

```diff
@@ -1961,6 +1961,38 @@ Tensor log1p_backward(const Tensor& grad, const Tensor& self) {
   return grad / (self + 1);
 }
 
+Tensor constant_pad_nd_backward(const Tensor& grad, const Tensor& self, IntList pad) {
+    auto l_inp = self.dim();
+    auto l_pad = pad.size() / 2;
+    auto l_diff = l_inp - l_pad;
+
+    auto grad_input = at::zeros(self.sizes(), grad.options());
+
+    auto cg_input = grad_input;
+    for (int i = l_diff; i < l_inp; i++) {
+        auto pad_idx = pad.size() - (i - l_diff + 1) * 2;
```

Comment:
This can be simplified to:
`pad_idx = 2 * (l_pad - i + l_diff - 1) = 2 * (l_inp - i - 1)`